zebra-chain/src/primitives/proofs/halo2.rs

```diff
@@ -22,15 +24,19 @@ impl fmt::Debug for Halo2Proof {
 
 impl ZcashSerialize for Halo2Proof {
     fn zcash_serialize<W: io::Write>(&self, mut writer: W) -> Result<(), io::Error> {
+        writer.write_compactsize(self.0.len() as u64)?;
         writer.write_all(&self.0[..])?;
```

Comment:
Looks like we need a `Vec<u8>::zcash_serialize` utility function, so it's easier to write code like this.

Reply:
Use `zcash_serialize_bytes` from https://github.com/ZcashFoundation/zebra/pull/2150 to simplify this function.